beacon_node/network/src/service.rs

```diff
@@ -111,19 +109,20 @@ impl<T: BeaconChainTypes> NetworkService<T> {
             propagation_percentage,
         };
 
-        let network_exit = runtime_handle.enter(|| spawn_service(network_service))?;
+        let _ = spawn_service(executor, network_service)?;
 
-        Ok((network_globals, network_send, network_exit))
+        Ok((network_globals, network_send))
     }
 }
 
 fn spawn_service<T: BeaconChainTypes>(
+    executor: environment::TaskExecutor,
     mut service: NetworkService<T>,
-) -> error::Result<tokio::sync::oneshot::Sender<()>> {
-    let (network_exit, mut exit_rx) = tokio::sync::oneshot::channel();
+) -> error::Result<()> {
+    let mut exit_rx = executor.exit();
```

Comment:
This is interesting. Does the exit fire inside this task before it is dropped from the global select?

Reply:
Yeah I have tested it out a bunch of times and it does fire and persist the DHT everytime I checked. Didn't think of the issue earlier. 
Thanks for pointing it out. Will check.

Reply:
@AgeManning  So I think this wasn't giving issues because the select in `TaskExecutor::spawn` takes futures in the order `task` and `exit`. So the future that's getting spawned has a preference over the exit future. When I changed the order in `select`, I stopped getting the DHT persisted logs. I'm not sure if there will be cases where the right future gets executed first when the exit future Signal fires. 
I have added a  `spawn_without_exit` function in task executor for cases where we want to do some additional cleanup after exit.